client/my-sites/invites/invite-accept-logged-in/index.jsx

```diff
@@ -29,10 +32,23 @@ export default React.createClass( {
 
 	accept() {
 		this.setState( { submitting: true } );
-		acceptInvite( this.props.invite );
-		store.set( 'invite_accepted', this.props.invite );
-		page( this.props.redirectTo );
 		analytics.tracks.recordEvent( 'calypso_invite_accept_logged_in_join_button_click' );
+		acceptInvite(
+			this.props.invite,
+			( error ) => {
+				if ( error ) {
+					this.setState( { submitting: false } );
+					if ( error.message ) {
+						this.props.errorNotice( error.message );
+					} else {
+						this.props.acceptErrororNotice( this.translate( 'There was an error accepting your invitation. Please try again.' ) )
+					}
+				} else {
```

Comment:
By using the callback we are going to loose all the optimistic magic. Maybe we should try to catch whatever could fail at form validation time, while possible?